### PR TITLE
BookA-Pset_BoreholeTestSPT & Pset_ChemicalLabTestLiquid

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_BoreholeTestSPT

## Description: 

Properties for results of Standard Penetration Tests/Standard Penetrometer Tests (SPTs).

| Property | Datatype | Description |
|----------|----------|-------------|
|NValue|IfcCountMeasure|N=N2+N3 is the result of the test.|
|YoungModulus|IfcModulusOfElasticityMeasure|According to existing definition.|

Managed to push Pset_ChemicalLabTestLiquid into the same PR:
# Pset: Pset_ChemicalLabTestLiquid

## Description: 

Properties for results of lab tests for chemical liquids.

| Property | Datatype | Description |
|----------|----------|-------------|
|EC|IfcReal|Electric conductivity. The reciprocal of the resistance, measured under specified conditions, between the opposite faces of a unit cube of defined dimensions of an aqueous solution. For water quality examination, this is often expressed as ""electrical conductivity"" and may be used as a measure of the concentration of ionizable solutes present in the sample.|
|PHValue|IfcReal|In chemistry, pH historically denoting ""potential of hydrogen"" (or ""power of hydrogen""), is a scale used to specify the acidity or basicity of an aqueous solution. Acidic solutions (solutions with higher concentrations of H+ ions) are measured to have lower pH values than basic or alkaline solutions. pH (potential of hydrogen) is the measure of the activity of hydrogen ions in solution. pH value is logarithm to the base 10 of the ratio of the molar hydrogen-ion activity (aH) multiplied by ?1|
|WaterTemperature|IfcThermodynamicTemperatureMeasure|Water temperature when testing.|
|Cl-|IfcReal|content of Cl- in the water sample.|
|NO3 -|IfcReal|content of NO3- in the water sample.|
|SO4 2-|IfcReal|content of SO4 2- in the water sample.|
|HCO3 -|IfcReal|content of HCO3- in the water sample.|
|Na+|IfcReal|content of Na+ in the water sample.|
|K +|IfcReal|content of K+ in the water sample.|
|Mg2+|IfcReal|content of Mg2+ in the water sample.|
|Ca2+|IfcReal|content of Ca2+ in the water sample.|

### CODE REVIEW

**Comment by patmclarin** [comment]
Please update description to: Properties for results of Standard Penetration Tests/Standard Penetrometer Tests (SPTs)

**Comment by larswik** [comment]
> Please update description to: Properties for results of Standard Penetration Tests/Standard Penetrometer Tests (SPTs)

Also updated the description of Pset_ChemicalLabTestLiquid to "Properties for results of lab tests for chemical liquids" to follow the same styleguide as for Pset_ChemicalLabTestSolid.

File: IFC4x3/Properties/c/Ca2+_10wmfjd5PB5xuqHQBOdisp/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Ca2+_10wmfjd5PB5xuqHQBOdisp" Name="Ca2+" UniqueId="40eb0a6d-9c56-4b17-be34-45a2d89ecdb3" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/c/Ca2+_10wmfjd5PB5xuqHQBOdisp/DocProperty.xml]: How about we rename this to "WaterHardness" or "CalciumContent"?

File: IFC4x3/Properties/c/Cl-_1AyulzLmb7kh4MInlhCAEO/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Cl-_1AyulzLmb7kh4MInlhCAEO" Name="Cl-" UniqueId="4af38bfd-5709-47ba-b116-4b1beb30a398" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/c/Cl-_1AyulzLmb7kh4MInlhCAEO/DocProperty.xml]: Similar to Ca2+ - "ChlorideContent"?

File: IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="EC_0JOFocHPD1Rxfnqm5NX9ca" Name="EC" UniqueId="1360fca6-4593-416f-ba71-d301578499a4" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/DocProperty.xml]: Propose "ElectricalConductivity" and datatype `IfcElectricConductanceMeasure`.

File: IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Electric conductivity. The reciprocal of the resistance, measured under specified conditions, between the opposite faces of a unit cube of defined dimensions of an aqueous solution. For water quality examination, this is often expressed as "electrical conductivity" and may be used as a measure of the concentration of ionizable solutes present in the sample.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/Documentation.md]: I would also take out the water specific stuff.
```suggestion
Electrical conductivity. Measured under specified conditions, between the opposite faces of a unit cube of defined dimensions of an aqueous solution. For water quality examination, this is often expressed as "electrical conductivity" and may be used as a measure of the concentration of ionizable solutes present in the sample.
```

File: IFC4x3/Properties/e/Emodulus_1Bctvx1L570B_eZVJbTFB8/DocProperty.xml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Emodulus_1Bctvx1L570B_eZVJbTFB8" Name="Emodulus" UniqueId="4b9b7e7b-0551-4700-bfa8-8df4e574f2c8" PrimaryDataType="IfcModulusOfElasticityMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/Emodulus_1Bctvx1L570B_eZVJbTFB8/DocProperty.xml]: Use `YoungModulus`.

File: IFC4x3/Properties/h/HCO3 -_1Ov1VDlGf0wwH8$gwlFai2/DocProperty.xml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HCO3 -_1Ov1VDlGf0wwH8$gwlFai2" Name="HCO3 -" UniqueId="58e417cd-bd0a-40eb-a448-feaeaf3e4b02" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/h/HCO3 -_1Ov1VDlGf0wwH8$gwlFai2/DocProperty.xml]: Propose "BicarbonateContent" or "Alcalinity".

File: IFC4x3/Properties/k/K +_2jm5TRU9TFhudeUCIbbCmE/DocProperty.xml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="K +_2jm5TRU9TFhudeUCIbbCmE" Name="K +" UniqueId="adc0575b-7897-4faf-89e8-78c4a594cc0e" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/k/K +_2jm5TRU9TFhudeUCIbbCmE/DocProperty.xml]: Propose "PotassiumContent".

File: IFC4x3/Properties/m/Mg2+_0IKcZxoxn5JhgLbQZ83Wbl/DocProperty.xml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Mg2+_0IKcZxoxn5JhgLbQZ83Wbl" Name="Mg2+" UniqueId="125268fb-cbbc-454e-ba95-95a8c80e096f" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/m/Mg2+_0IKcZxoxn5JhgLbQZ83Wbl/DocProperty.xml]: "MagnesiumContent"

File: IFC4x3/Properties/n/NO3 -_2mojFkIZf0RgDdRrAmsAvG/DocProperty.xml [hunk 9]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NO3 -_2mojFkIZf0RgDdRrAmsAvG" Name="NO3 -" UniqueId="b0cad3ee-4a3a-406e-a367-6f52b0d8ae50" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NO3 -_2mojFkIZf0RgDdRrAmsAvG/DocProperty.xml]: "NitrateContent"

File: IFC4x3/Properties/n/NValue_0ODh2KpO5Awv4vDj6gcncD/Documentation.md [hunk 10]
```diff
@@ -0,0 +1 @@
+N=N2+N3 is the result of the test.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NValue_0ODh2KpO5Awv4vDj6gcncD/Documentation.md:1]: But what is `NValue`? I would expect a description of `NValue` in the description of the NValue property. Please provide a better description.

File: IFC4x3/Properties/n/Na+_1yejeoaXj1gAsLahNn8Fv1/DocProperty.xml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Na+_1yejeoaXj1gAsLahNn8Fv1" Name="Na+" UniqueId="7ca2da32-921b-41a8-ad95-92b5f120fe41" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/Na+_1yejeoaXj1gAsLahNn8Fv1/DocProperty.xml]: NatriumContent

File: IFC4x3/Properties/p/PHValue_0447Chz7H82wDEEo6Q3khK/DocProperty.xml [hunk 12]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PHValue_0447Chz7H82wDEEo6Q3khK" Name="PHValue" UniqueId="0410732b-f474-480b-a34e-3b219a0eead4" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PHValue_0447Chz7H82wDEEo6Q3khK/DocProperty.xml]: Use `PHLevel`.

File: IFC4x3/Properties/s/SO4 2-_0U5FAS_db4URK3Zf0YMJDR/DocProperty.xml [hunk 13]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SO4 2-_0U5FAS_db4URK3Zf0YMJDR" Name="SO4 2-" UniqueId="1e14f29c-fa79-4479-b503-8e902259335b" PrimaryDataType="IfcReal" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/SO4 2-_0U5FAS_db4URK3Zf0YMJDR/DocProperty.xml]: SuplhateContent

File: IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml [hunk 14]
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="WaterTemperature_1K5VCc3ND6CQln9UD4cMgu" Name="WaterTemperature" UniqueId="5415f326-0d73-4631-abf1-25e344996ab8" PrimaryDataType="IfcThermodynamicTemperatureMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml]: Use existing `Temperature`.

**Review by SergejMuhic — COMMENTED** [review]
I would propose a meeting on this pset. Quite a lot to unpack.

Please consider Pset_MaterialWater for Pset_ChemicalLabTestLiquid.

I would not use `IfcReal` for the "Content" properties. Use `IfcAmountOfSubstanceMeasure` or `IfcIonConcentrationMeasure`.

File: IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml (hunk 14 above)
Comment (larswik) [reply IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml]: > Use existing `Temperature`.

Maybe this means loosing semantics?

Comment (SergejMuhic) [reply IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml]: Most likely. The pset is probably not exclusively for water, right? The properties do imply only water testing though.

File: IFC4x3/Properties/e/Emodulus_1Bctvx1L570B_eZVJbTFB8/DocProperty.xml (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_1Bctvx1L570B_eZVJbTFB8/DocProperty.xml]: Fixed in [78d1f16](https://github.com/bSI-InfraRoom/IFC-Specification/pull/597/commits/78d1f1672489922a8eb8773bdb382784e4071e42)

File: IFC4x3/Properties/c/Ca2+_10wmfjd5PB5xuqHQBOdisp/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/c/Ca2+_10wmfjd5PB5xuqHQBOdisp/DocProperty.xml]: According to discussions with domain experts, we keep the properties from Pset_ChemicalLabTestLiquid in Pset_MaterialWater. Not certain that WaterHardness and CalciumContent refers to the same.

File: IFC4x3/Properties/c/Cl-_1AyulzLmb7kh4MInlhCAEO/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/c/Cl-_1AyulzLmb7kh4MInlhCAEO/DocProperty.xml]: No. See WaterHardness above.

File: IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/DocProperty.xml]: Done.

File: IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/Documentation.md (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/e/EC_0JOFocHPD1Rxfnqm5NX9ca/Documentation.md]: Done.

File: IFC4x3/Properties/e/Emodulus_1Bctvx1L570B_eZVJbTFB8/DocProperty.xml (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_1Bctvx1L570B_eZVJbTFB8/DocProperty.xml]: Done.

File: IFC4x3/Properties/n/NValue_0ODh2KpO5Awv4vDj6gcncD/Documentation.md (hunk 10 above)
Comment (larswik) [reply IFC4x3/Properties/n/NValue_0ODh2KpO5Awv4vDj6gcncD/Documentation.md:1]: NValue removed.

File: IFC4x3/Properties/n/Na+_1yejeoaXj1gAsLahNn8Fv1/DocProperty.xml (hunk 11 above)
Comment (larswik) [reply IFC4x3/Properties/n/Na+_1yejeoaXj1gAsLahNn8Fv1/DocProperty.xml]: Keeping the names proposed by the domain experts.

File: IFC4x3/Properties/p/PHValue_0447Chz7H82wDEEo6Q3khK/DocProperty.xml (hunk 12 above)
Comment (larswik) [reply IFC4x3/Properties/p/PHValue_0447Chz7H82wDEEo6Q3khK/DocProperty.xml]: Done.

File: IFC4x3/Properties/s/SO4 2-_0U5FAS_db4URK3Zf0YMJDR/DocProperty.xml (hunk 13 above)
Comment (larswik) [reply IFC4x3/Properties/s/SO4 2-_0U5FAS_db4URK3Zf0YMJDR/DocProperty.xml]: Keeping the names proposed by domain experts.

File: IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml (hunk 14 above)
Comment (larswik) [reply IFC4x3/Properties/w/WaterTemperature_1K5VCc3ND6CQln9UD4cMgu/DocProperty.xml]: Done.

**Pull request merged**